### DISCUSSION TITLE
List plugins to be loaded across multiple files

### DISCUSSION THREAD

**Discussion by ghost** [wbthomason/packer.nvim#277] [Ideas]
To use an emacs analogy, since packer is inspired by use-package, Doom emacs includes many modules that can be enabled by users uncommenting lines in an init file. Each module's directory includes a file with packages (i.e. in `modules/lang/cc/packages.el`) that are added to the list of all packages to install. How would one go about using packer in a similar manner by getting lists of package configurations from multiple files?

**Comment by wbthomason** [comment]
This seems closely tied to the discussion of the proposed `requirements.lua` format (#176). Do you think that would address this need?

Alternatively, if you already have the lists of plugins, you can absolutely spread your calls to `use` across multiple files. Just be sure that `packer.init()` is called before the first invocation and that all the relevant files with `use` invocations get sourced before you run any operations like `install`, `update`, `sync`, etc.

**Reply by ghost** [reply]
> Alternatively, if you already have the lists of plugins, you can absolutely spread your calls to use across multiple files. Just be sure that packer.init() is called before the first invocation and that all the relevant files with use invocations get sourced before you run any operations like install, update, sync, etc.

That's mostly what I was looking for. I will try to get it working on my own (learning Lua as I go), and then I might have some follow up questions about overriding existing `use` invocations in files that get loaded later.

**Comment by ghost** [comment]
For posterity, @glepnir has a [neovim config](https://github.com/glepnir/nvim) that is a helpful example of distributing the `use` invocations to multiple files.
